Add support for OpenAI responses API

the-gigi opened the pull request:
Adding support to the [OpenAI responses API](https://platform.openai.com/docs/api-reference/responses)
See https://github.com/sashirestela/simple-openai/issues/307

- Building on the domain classes added by @sashirestela 
- Add additional domain classes
- Add the services to OpenAI and SimpleOpenAI
- Add unit tests
- Add comprehensive demo
- Update the README (except the image of the Google sheet with the endpoints)

During the work I also added some resources and discovered they miss a newline in the end. I fixed it with a refactoring, which added a newline to many other resource files. I hope it's OK.

the-gigi committed:
Add support for OpenAI responses API

the-gigi commented:
@sashirestela  spotless check fail, but it passes on my machine 🤷 

![image](https://github.com/user-attachments/assets/25864127-1ccf-4d3b-8a27-5177d68022dc)

sashirestela commented:
@the-gigi, thanks for your PR. There are previous things that you need to do before I could go on a deep revision. Below are my first observations:

1. Focus on implementing the new functionality, don't add other changes not related to the main topic. 
2. In the [failing build](https://github.com/sashirestela/simple-openai/actions/runs/14830317560/job/41630069518?pr=322) you can see the detail of the 9 files with format violations.
3. You need to sync your branch with the latest changes that I uploaded to the [307-new-responses-api](https://github.com/sashirestela/simple-openai/tree/307-new-responses-api) branch one week ago. That was necessary because the domain classes were incomplete.

the-gigi committed:
fix spotless errors, by temporarily commenting out <ratchetFrom>origin/main</ratchetFrom> in pom.xml

the-gigi committed:
remove newline at the end of json files

the-gigi committed:
updated chat domain service tier

the-gigi committed:
remove newline at the end of json files properly

the-gigi committed:
remove newline from another json file

the-gigi committed:
added input test

the-gigi committed:
add ResponseTextTest

the-gigi committed:
add more tests

the-gigi committed:
more tests

the-gigi committed:
additional tests + better test coverage

the-gigi commented:
> @the-gigi, thanks for your PR. There are previous things that you need to do before I could go on a deep revision. Below are my first observations:
> 
> 1. Focus on implementing the new functionality, don't add other changes not related to the main topic.

Sounds good. I reverted the newline at the end of file changes.

> 2. In the [failing build](https://github.com/sashirestela/simple-openai/actions/runs/14830317560/job/41630069518?pr=322) you can see the detail of the 9 files with format violations.

I figured it out. It looks like between forking from a branch, rebasing and squashing commits that spotless skipped checking/applying for the new files. To fix it, I had to temporarily comment out this in pom.xml (don't worry. not committed)
```
      <plugin>
        <groupId>com.diffplug.spotless</groupId>
        <artifactId>spotless-maven-plugin</artifactId>
        <version>${spotless.version}</version>
        <configuration>
          <!--ratchetFrom>origin/main</ratchetFrom -->  👈 
```

> 3. You need to sync your branch with the latest changes that I uploaded to the [307-new-responses-api](https://github.com/sashirestela/simple-openai/tree/307-new-responses-api) branch one week ago. That was necessary because the domain classes were incomplete.

That's too bad. When I took over and started from your original [[307-new-responses-api]](https://github.com/sashirestela/simple-openai/tree/307-new-responses-api) branch I thought I'll work on it until the PR. I didn't realize you keep working on it too. So, I continued your work on the domain. Your additional changes are mostly incompatible with the current state of the PR (except for the Reasoning class where we are compatible). I added the unrelated change to ChatRequest.java. I'll see how complicated it is to integrate your domain changes, but if it's not trivial I might not have the time in the next couple of months (work + vacation).

sashirestela commented:
> That's too bad. When I took over and started from your original [[307-new-responses-api]](https://github.com/sashirestela/simple-openai/tree/307-new-responses-api) branch I thought I'll work on it until the PR. I didn't realize you keep working on it too. So, I continued your work on the domain. Your additional changes are mostly incompatible with the current state of the PR (except for the Reasoning class where we are compatible). I added the unrelated change to ChatRequest.java. I'll see how complicated it is to integrate your domain changes, but if it's not trivial I might not have the time in the next couple of months (work + vacation).

Sadly, some domain classes were pending. Moreover, OpenAI is committing domain changes even these days. It seems that the Response API is quite unstable yet. I'm doing a final revision to domain classes today and I'm identifying some changes.

sashirestela commented:
Another PR is in progress to include the latest domain changes from OpenAI. Thanks anyway @the-gigi

the-gigi commented:
> Another PR is in progress to include the latest domain changes from OpenAI. Thanks anyway @the-gigi

No problem. I'm always happy to help when I can.
